src/pages/login/CertificateGenerateBtn.tsx

```diff
@@ -0,0 +1,95 @@
+import type { FC } from "react";
+import { useEffect, useState } from "react";
+import PasswordModal from "./PasswordModal";
+import { Button, Icon } from "@canonical/react-components";
+
+interface Certs {
+  crt: string;
+  pfx: string;
+}
+
+const CertificateGenerateBtn: FC = () => {
+  const [isGenerating, setGenerating] = useState(false);
+  const [isModalOpen, setModalOpen] = useState(false);
+  const [certs, setCerts] = useState<Certs | null>(null);
+
+  const closeModal = () => {
+    setModalOpen(false);
+  };
+
+  const openModal = () => {
+    setModalOpen(true);
+  };
+
+  const sendPfx = () => {
+    if (certs) {
+      downloadBase64(`lxd-ui-${location.hostname}.pfx`, certs.pfx);
+    }
+  };
+
+  useEffect(() => {
+    if (certs) {
+      sendPfx();
+    }
+  }, [certs]);
+
+  const createCert = (password: string) => {
+    closeModal();
+    setGenerating(true);
+
+    const worker = new Worker(
+      new URL("../../util/certificate?worker", import.meta.url),
+      { type: "module" },
+    );
+
+    worker.onmessage = (event: MessageEvent<Certs>) => {
+      setCerts(event.data);
+      setGenerating(false);
+      worker.terminate();
+    };
+
+    worker.onerror = (error) => {
+      console.error("Web Worker error:", error);
+      setGenerating(false);
+      worker.terminate();
+    };
+
+    worker.postMessage(password);
+  };
```

Comment:
```suggestion

  const createCert = (password: string) => {
    closeModal();
    setGenerating(true);

    const worker = new Worker(
      new URL("../../util/certificate?worker", import.meta.url),
      { type: "module" },
    );

    worker.onmessage = (event: MessageEvent<Certs>) => {
      setCerts(event.data);
      setGenerating(false);
      downloadBase64(`lxd-ui-${location.hostname}.pfx`, event.data.pfx);
      worker.terminate();
    };

    worker.onerror = (error) => {
      console.error("Web Worker error:", error);
      setGenerating(false);
      worker.terminate();
    };

    worker.postMessage(password);
  };
```
We can make this slightly more straight forward now.